go/rtl/auth.go

```diff
@@ -123,16 +128,24 @@ func (s *AuthSession) Do(result interface{}, method, ver, path string, reqPars m
 		}
 	}
 
-	// create request context with timeout
-	var timeoutInSeconds int32 = 120 //seconds
+	parent := context.Background()
+	if s.Config.Context != nil {
+		parent = s.Config.Context
+	}
+	if options != nil && options.Context != nil {
+		parent = options.Context
+	}
+
+	// create request context with timeout from options or else config or else 120 seconds
+	var timeoutInSeconds int32 = 120
 	if s.Config.Timeout != 0 {
 		timeoutInSeconds = s.Config.Timeout
 	}
 	if options != nil && options.Timeout != 0 {
 		timeoutInSeconds = options.Timeout
 	}
 
-	ctx, cncl := context.WithTimeout(context.Background(), time.Second*time.Duration(timeoutInSeconds))
+	ctx, cncl := context.WithTimeout(parent, time.Second*time.Duration(timeoutInSeconds))
```

Comment:
If the parent context has a different timeout, what's the expected behavior here?

Reply:
Not exactly. If the parent hits its cancellation condition (e.g. a timeout) first then the cancellation propagates to the child.  If the child cancels it depends on how that specific error is handled... perhaps the request retries, perhaps the error is passed to the calling function, in which case most of the time the function fails and the parent context then cancels, which propagates the cancellation to all other goroutines, etc etc.

Basically this approach means you can have an overall timeout set by a parent, but each individual request running in a goroutine can also have its own timeout... this avoids the scenario where you set a 60min overall timeout, but one bad request has to wait the entire 60m to time out and fail.